civihr_employee_portal/views/includes/views_between_dates_filter_handler.inc

```diff
@@ -60,26 +76,48 @@ class views_between_dates_filter_handler extends date_views_filter_handler_simpl
     // SQL: start date <= value AND end date >= value.
     // If data is correct.
     if (count($field_names) == 2) {
-      $value = $this->get_filter_value('value', $this->value['value']);
+      $value = $this->get_date_value_formatted($this->get_filter_value('value', $this->value['value']));
       $comp_date = new DateObject($value, date_default_timezone(), $this->format);
-
       $start_field = $this->date_handler->sql_field($field_names[0], NULL, $comp_date);
       $start_field = $this->date_handler->sql_format($this->format, $start_field);
       $end_field = $this->date_handler->sql_field($field_names[1], NULL, $comp_date);
       $end_field = $this->date_handler->sql_format($this->format, $end_field);
       $placeholder = $this->placeholder();
       $group = !empty($this->options['date_group']) ? $this->options['date_group'] : $this->options['group'];
       if ($this->operator == '=') {
-        $this->query->add_where_expression($group, "$start_field <= $placeholder AND $end_field >= $placeholder", array($placeholder => $value));
+        if ((int)$this->options['include_null_dates'] === 2) {
+          $this->query->add_where_expression($group, "($start_field <= $placeholder OR $start_field IS NULL) AND ($end_field >= $placeholder OR $end_field IS NULL)", array($placeholder => $value));
+        } else {
+          $this->query->add_where_expression($group, "$start_field <= $placeholder AND $end_field >= $placeholder", array($placeholder => $value));
+        }
       }
       // Else not between
       else {
         $this->query->add_where_expression($group, "$start_field > $placeholder OR $end_field < $placeholder", array($placeholder => $value));
       }
    }
+  }
 
+  /**
+   * Get date value formatted according to date format set up by input field.
```

Comment:
Can you elaborate this comment a little more? What is the date format set up by the input field? Why is the original date returned if $parts == 1?
Is not clear from and to what format the dates are converted. Assuming the original date is in dd/mm/yyyy format, what about changing the code on lines 108 to something like?:
```php
list($day, $month, $year) = explode('/', $date);
```